gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/FSSpecStore.java

```diff
@@ -206,23 +232,32 @@ public Spec getSpec(URI specUri, String version) throws IOException, SpecNotFoun
     return readSpecFromFile(specPath);
   }
 
-  @Override
-  public Collection<Spec> getAllVersionsOfSpec(URI specUri) throws IOException, SpecNotFoundException {
-    Preconditions.checkArgument(null != specUri, "Spec URI should not be null");
-
-    Collection<Spec> specs = getSpecs();
-    Collection<Spec> filteredSpecs = Lists.newArrayList();
-    for (Spec spec : specs) {
-      if (spec.getUri().equals(specUri)) {
-        filteredSpecs.add(spec);
+  private Collection<Spec> getAllVersionsOfSpec(String specGroup, String specName) throws IOException {
+    Collection<Spec> specs = Lists.newArrayList();
+    FileStatus[] fileStatuses;
+    try {
+      if (StringUtils.isEmpty(specGroup)) {
```

Comment:
Can you create a private method that returns a path based on the spec group and root dir? Then the listStatus does not have to be duplicated. There is another use like this above.